inference-engine/tests/functional/plugin/myriad/subgraph_tests/dsr_gather.cpp

```diff
@@ -2,142 +2,159 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+#include "dsr_tests_common.hpp"
+
 #include <functional_test_utils/layer_test_utils.hpp>
 #include <ngraph_functions/builders.hpp>
 #include <vpu/ngraph/operations/dynamic_shape_resolver.hpp>
 
 namespace {
 
-using DataType = ngraph::element::Type_t;
-using DataDims = ngraph::Shape;
+using namespace LayerTestsUtils::vpu;
 
+const std::vector<ngraph::element::Type> dataTypeVector = {
+        ngraph::element::f16,
+        ngraph::element::f32,
+        ngraph::element::i32,
+};
+
+const std::vector<ngraph::element::Type> idxTypeVector = {
+        ngraph::element::i32,
+};
 
 struct GatherTestCase {
-    ngraph::Shape data_shape, index_shape;
-    int64_t axis, first_split_point, second_split_point;
+    DataShapeWithUpperBound inputShapes;
+    DataShapeWithUpperBound indexShape;
+    int64_t axis, firstSplitPoint, secondSplitPoint;
 };
 
-const auto combinations = testing::Combine(
-    testing::Values(
-            ngraph::element::f16,
-            ngraph::element::f32,
-            ngraph::element::i32,
-            ngraph::element::i64,
-            ngraph::element::u8),
-    testing::Values(
-            ngraph::element::i32,
-            ngraph::element::i64,
-            ngraph::element::u8),
-    testing::Values(
-            GatherTestCase{{6}, {15, 4, 20, 28}, 0, 0, 0},
-            GatherTestCase{{6, 12, 10, 24}, {6}, 0, 0, 1},
-            GatherTestCase{{6, 12}, {15, 4, 20, 28}, 1, 1, 2},
-            GatherTestCase{{6, 12, 10, 24}, {15, 4, 20, 28}, 3, 3, 4},
-            GatherTestCase{{6, 12, 10, 24}, {15, 4, 20, 28}, -1, 3, 4},
-            GatherTestCase{{6, 12, 10, 24}, {15, 4, 20, 28}, -4, 0, 1}),
-    testing::Values(CommonTestUtils::DEVICE_MYRIAD));
-
-
-using Parameters = std::tuple<
+using GatherParameters = std::tuple<
     DataType,
     DataType,
     GatherTestCase,
     LayerTestsUtils::TargetDevice
 >;
 
-class DSR_GatherData : public testing::WithParamInterface<Parameters>,
-        public LayerTestsUtils::LayerTestsCommon {
+class DSR_GatherBase : public testing::WithParamInterface<GatherParameters>,
+                       public DSR_TestsCommon {
+protected:
+    InferenceEngine::Blob::Ptr GenerateInput(const InferenceEngine::InputInfo& info) const override {
+        const auto& name = info.name();
+        const auto suffix = std::string("/indices");
+        if (name.compare(name.length() - suffix.length(), suffix.length(), suffix) == 0) {
+            const auto& parameters = GetParam();
+            const auto& gatherSetup = std::get<2>(parameters);
+            const auto& inputRank = gatherSetup.inputShapes.shape.size();
+            const auto axis = gatherSetup.axis < 0 ? gatherSetup.axis + inputRank : gatherSetup.axis;
+
+            const auto startValue = 0;
+            const auto endValue = gatherSetup.inputShapes.shape[axis] - 1;
+
+            return FuncTestUtils::createAndFillBlob(info.getTensorDesc(), endValue - startValue, startValue);
```

Comment:
Don't make tests more complicated that it's really needed. Just hardcode `0` instead of `startValue`.
Do you generate values in range [0, endValue) with step 1 here?

Reply:
Done